MCMC/utils.py

```diff
@@ -1,33 +1,32 @@
 import freud
 import numpy as np
+import gsd.hoomd
 from numba import jit
 
-def avg_nn(sim_obj, n_frames, r_max=3):
-    box = [sim_obj.L, sim_obj.L,0]
-    frame_avg_nn = []
-    for points in sim_obj.system_history[-n_frames:]:
-        points = np.append(
-            points, np.zeros((points.shape[0], 1)),axis=1
-        )
+def avg_nn(gsd_file, frame=-1, r_max=3):
+    with gsd.hoomd.open(gsd_file) as f:
+        snap = f[frame]
+        box = snap.configuration.box
+        points = snap.particles.position
+        frame_avg_nn = []
         aq = freud.locality.AABBQuery(box, points)
         nlist = aq.query(points, {'r_max': 3}).toNeighborList()
         neighbors = []
-        for i in range(sim_obj.n_particles):
+        for i in range(snap.particles.N):
             neighbors.append(np.where(nlist[:, 0] == i)[0].shape[0]-1)
             frame_avg_nn.append(np.average(neighbors))
     return np.average(frame_avg_nn)
 
-def structure_factor(sim_obj, n_frames, num_k_values=100, k_max=10):
-    box = [sim_obj.L, sim_obj.L,0]
-    sf = freud.diffraction.StaticStructureFactorDebye(
+def structure_factor(gsd_file, frame=-1, num_k_values=100, k_max=10):
+    with gsd.hoomd.open(gsd_file) as f:
+        snap = f[frame]
```

Comment:
If I'm not mistaken, I think we can use the structure factor module in freud in a similar way that we do it with RDFs in cmeutils in the sense that we can pass in a range of frames (start, finish) and get a structure factor that is averaged over these frames.  If you look at the `gsd_rdf` function in `structure.py` you can see how we do it there.